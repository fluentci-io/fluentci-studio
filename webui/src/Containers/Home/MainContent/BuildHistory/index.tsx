import BuildHistory from "./BuildHistory";

export default BuildHistory;
