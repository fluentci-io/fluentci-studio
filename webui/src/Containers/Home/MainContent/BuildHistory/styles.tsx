import { css } from "@emotion/react";
import styled from "@emotion/styled";

export const Container = styled.div`
  position: relative;
  display: flex;
  align-items: center;
  justify-content: space-between;
  flex-direction: row;
`;

export const Bars = styled.div`
  display: flex;
  flex: 1;
  height: 40px;
  justify-content: center;
  margin-right: 10px;
`;

export const Info = styled.div`
  height: 50px;
  color: #7392b1;
  font-size: 12px;
  width: 100px;
`;

export const Value = styled.div`
  font-size: 24px;
  color: #fff;
`;

export const Unit = styled.span`
  font-size: 12px;
  color: #7392b1;
  margin-left: 5px;
`;

export const Status = styled.div<{ status?: string }>`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 80px;
  width: 50px;
  border-radius: 5px;

  ${({ status }) =>
    status === "SUCCESS" &&
    css`
      background-color: #05f07719;
    `}
  ${({ status }) =>
    status === "FAILURE" &&
    css`
      background-color: #ff006a13;
    `}
    ${({ status }) =>
    status === "PENDING" &&
    css`
      background-color: #ffa60020;
    `}
`;

export const Bar = styled.div<{ max?: number; n?: number; success?: boolean }>`
  height: 0%;
  width: 5px;
  background-color: #ff006a;
  position: absolute;
  border-top-left-radius: 3px;
  border-top-right-radius: 3px;

  left: 0;
  bottom: 0;
  ${({ max }) =>
    max &&
    css`
      height: ${max}%;
    `}
  ${({ n }) =>
    n &&
    css`
      left: ${5 * n + 1 * n}px;
    `}
  ${({ success }) =>
    success === true &&
    css`
      background-color: #05f076;
    `}
`;
