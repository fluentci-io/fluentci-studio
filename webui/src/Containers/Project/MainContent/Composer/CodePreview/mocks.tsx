export const code = `name: lint and test
on:
  push:
    branches:
      - main
jobs:
  tests:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: Setup Fluent CI
        uses: fluentci-io/setup-fluentci@v5
        with:
          wasm: true
          pipeline: deno
          args: |
            fmt
            lint
            test
            coverage
      - name: Upload to Codecov
        run: fluentci run --wasm codecov upload
        env:
          CODECOV_TOKEN: \${{ secrets.CODECOV_TOKEN }}`;
