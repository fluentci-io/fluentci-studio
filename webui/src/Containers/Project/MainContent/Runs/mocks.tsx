export const runs = [
  {
    id: "edggjsrj42ysd3gljsksn4tc",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:22:57.031Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 41185,
    jobs: [
      {
        id: "ycsbwbbi0rvlh5mnsjb7lp0p",
        name: "test",
        createdAt: "2024-05-30T14:22:57.017Z",
        status: "SUCCESS",
        duration: 23847,
        __typename: "Job",
      },
      {
        id: "maq4cxhvyirvvzx19xxwzbe5",
        name: "build",
        createdAt: "2024-05-30T14:22:57.017Z",
        status: "SUCCESS",
        duration: 16992,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #20",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "mwf27469282wpoilujzzsfag",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:20:54.866Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 68358,
    jobs: [
      {
        id: "hrf5wd9agnkic1hapd6r94vp",
        name: "test",
        createdAt: "2024-05-30T14:20:54.860Z",
        status: "SUCCESS",
        duration: 33102,
        __typename: "Job",
      },
      {
        id: "if8ntcrxe71sqfbf2v8oojhk",
        name: "build",
        createdAt: "2024-05-30T14:20:54.860Z",
        status: "SUCCESS",
        duration: 34942,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #19",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "q0xrwtha16ayq2c3z28dvc9i",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:19:17.866Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 39889,
    jobs: [
      {
        id: "v83f464e4rn42o90m2uh8pa3",
        name: "test",
        createdAt: "2024-05-30T14:19:17.851Z",
        status: "SUCCESS",
        duration: 20635,
        __typename: "Job",
      },
      {
        id: "pz5xq17z4ba1n7isl6de2031",
        name: "build",
        createdAt: "2024-05-30T14:19:17.851Z",
        status: "SUCCESS",
        duration: 18890,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #18",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "amzu16be4ewpjd0a9znyuotj",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:18:16.439Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 55607,
    jobs: [
      {
        id: "moog6z4srrcu3fwjx2uay5ln",
        name: "test",
        createdAt: "2024-05-30T14:18:16.425Z",
        status: "SUCCESS",
        duration: 23039,
        __typename: "Job",
      },
      {
        id: "g6t8p941shlglf3p13qcwmxu",
        name: "build",
        createdAt: "2024-05-30T14:18:16.425Z",
        status: "SUCCESS",
        duration: 32169,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #17",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "x4e4om30gsrlxh1m3ky21u1x",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:17:27.947Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 42322,
    jobs: [
      {
        id: "lwmvsfu72epflue66uo9zopn",
        name: "test",
        createdAt: "2024-05-30T14:17:27.933Z",
        status: "SUCCESS",
        duration: 21664,
        __typename: "Job",
      },
      {
        id: "s1nhlyb29qx83xkr6p5mg1c7",
        name: "build",
        createdAt: "2024-05-30T14:17:27.933Z",
        status: "SUCCESS",
        duration: 20347,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #16",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "kzhijmkhcwr6lx5h2jnd11zr",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:15:39.535Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 50528,
    jobs: [
      {
        id: "gsq0y27j78lt2lhnw9vgmbg2",
        name: "test",
        createdAt: "2024-05-30T14:15:39.521Z",
        status: "SUCCESS",
        duration: 32533,
        __typename: "Job",
      },
      {
        id: "k5a5vdlvjdgutdwotydt4jk9",
        name: "build",
        createdAt: "2024-05-30T14:15:39.521Z",
        status: "SUCCESS",
        duration: 17566,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #15",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "at65tgplme2l69zfe6n7ud64",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:14:35.948Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 23565,
    jobs: [
      {
        id: "dhzrqi7qjni4tbn88a9eo89h",
        name: "test",
        createdAt: "2024-05-30T14:14:35.934Z",
        status: "SUCCESS",
        duration: 16636,
        __typename: "Job",
      },
      {
        id: "qeilrmh0t10hp5nycxxb4upz",
        name: "build",
        createdAt: "2024-05-30T14:14:35.934Z",
        status: "SUCCESS",
        duration: 6701,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #14",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "ku7usad1ep81ow38hy24wz73",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:13:00.412Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 34670,
    jobs: [
      {
        id: "ntzd8zzu2oxu39rkah4akqqw",
        name: "test",
        createdAt: "2024-05-30T14:13:00.404Z",
        status: "SUCCESS",
        duration: 16988,
        __typename: "Job",
      },
      {
        id: "nrwm0axjzbymtru46a53yl8o",
        name: "build",
        createdAt: "2024-05-30T14:13:00.405Z",
        status: "SUCCESS",
        duration: 17399,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #13",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "hx096gahj4t9woffp6tcwdim",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:12:05.348Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 46296,
    jobs: [
      {
        id: "hug3tcftru030qi6r3sztdjj",
        name: "test",
        createdAt: "2024-05-30T14:12:05.340Z",
        status: "SUCCESS",
        duration: 26562,
        __typename: "Job",
      },
      {
        id: "br8obw3kw2l7w1c5ruucc5ff",
        name: "build",
        createdAt: "2024-05-30T14:12:05.341Z",
        status: "SUCCESS",
        duration: 19454,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #12",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "ld7ihjoovu8a65v75w6edrh3",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:10:03.192Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 70604,
    jobs: [
      {
        id: "bpy78tg6gqwkh2paj8ndlsqo",
        name: "test",
        createdAt: "2024-05-30T14:10:03.174Z",
        status: "SUCCESS",
        duration: 48982,
        __typename: "Job",
      },
      {
        id: "j004pg9xb7xla863okdhn1mo",
        name: "build",
        createdAt: "2024-05-30T14:10:03.174Z",
        status: "SUCCESS",
        duration: 21308,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #11",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "m4hynmr7rjm30n8ycq600fbh",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:08:32.771Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 43102,
    jobs: [
      {
        id: "tu8iey7rtzua4b05w3z7g32v",
        name: "test",
        createdAt: "2024-05-30T14:08:32.757Z",
        status: "SUCCESS",
        duration: 20721,
        __typename: "Job",
      },
      {
        id: "lncvyej458t6lsa7en4kpguz",
        name: "build",
        createdAt: "2024-05-30T14:08:32.757Z",
        status: "SUCCESS",
        duration: 21931,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #10",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "rxi3t2u71belkif6qsi0f2l0",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:06:47.122Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 39296,
    jobs: [
      {
        id: "ruqzw3jg76ur2sg70jpj2tbc",
        name: "test",
        createdAt: "2024-05-30T14:06:47.049Z",
        status: "SUCCESS",
        duration: 26299,
        __typename: "Job",
      },
      {
        id: "nickylgy2zx63o4glj810uaq",
        name: "build",
        createdAt: "2024-05-30T14:06:47.049Z",
        status: "SUCCESS",
        duration: 12549,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #9",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "rxtfxrc8m17tk8oqmpbno5ew",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:05:40.260Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 83047,
    jobs: [
      {
        id: "qwadak7vto8n738rteszzat8",
        name: "test",
        createdAt: "2024-05-30T14:05:40.244Z",
        status: "SUCCESS",
        duration: 54105,
        __typename: "Job",
      },
      {
        id: "zwqrig2ih005i57x5uag6nof",
        name: "build",
        createdAt: "2024-05-30T14:05:40.245Z",
        status: "SUCCESS",
        duration: 28416,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #8",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "xaojk56w04i7eawug9q4yoai",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:04:17.514Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 73309,
    jobs: [
      {
        id: "h2ju8fx6bo3e3bfvmxd1hszt",
        name: "test",
        createdAt: "2024-05-30T14:04:17.500Z",
        status: "SUCCESS",
        duration: 46859,
        __typename: "Job",
      },
      {
        id: "s03i79rajqisoacdjnf1wx9s",
        name: "build",
        createdAt: "2024-05-30T14:04:17.500Z",
        status: "SUCCESS",
        duration: 26013,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #7",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "qljan7aqkbygb52xol5vjpp8",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:03:21.397Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 44673,
    jobs: [
      {
        id: "v2sct9q5ly2gcfokhjklfxpd",
        name: "test",
        createdAt: "2024-05-30T14:03:21.382Z",
        status: "SUCCESS",
        duration: 25699,
        __typename: "Job",
      },
      {
        id: "xvaat87df5z1qzw7865zkj4v",
        name: "build",
        createdAt: "2024-05-30T14:03:21.382Z",
        status: "SUCCESS",
        duration: 18621,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #6",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "rn7geppeqb3c294qwwcn8wnn",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:02:02.521Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 51136,
    jobs: [
      {
        id: "z14dc7y19l2dnp3i3zwig6xt",
        name: "test",
        createdAt: "2024-05-30T14:02:02.515Z",
        status: "SUCCESS",
        duration: 29452,
        __typename: "Job",
      },
      {
        id: "ctz2elloaqw9ll17v7zgj18t",
        name: "build",
        createdAt: "2024-05-30T14:02:02.515Z",
        status: "SUCCESS",
        duration: 20975,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #5",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "cqr74vyuxmlxb7yerhfcgv14",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:00:06.821Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 63649,
    jobs: [
      {
        id: "p87s1b1xlt7wykvtvgvecow0",
        name: "test",
        createdAt: "2024-05-30T14:00:06.805Z",
        status: "SUCCESS",
        duration: 34409,
        __typename: "Job",
      },
      {
        id: "b7decu7kigqarxpeh0oru4f3",
        name: "build",
        createdAt: "2024-05-30T14:00:06.805Z",
        status: "SUCCESS",
        duration: 28703,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #4",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "emo4b7cvjvldxzigd357qi41",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T13:58:34.972Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 82346,
    jobs: [
      {
        id: "drkgjqc4l43fwzj6fldy0a5p",
        name: "test",
        createdAt: "2024-05-30T13:58:34.964Z",
        status: "SUCCESS",
        duration: 48136,
        __typename: "Job",
      },
      {
        id: "ec6zixy9biisx54z7h64bfz0",
        name: "build",
        createdAt: "2024-05-30T13:58:34.964Z",
        status: "SUCCESS",
        duration: 33622,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #3",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "qj019uwyx55v8z74yp36dsxn",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-28T15:32:19.489Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 17420,
    jobs: [
      {
        id: "ik49e580m6vpr8qi4c6x94mj",
        name: "test",
        createdAt: "2024-05-28T15:32:19.489Z",
        status: "SUCCESS",
        duration: 7986,
        __typename: "Job",
      },
      {
        id: "jcf212n7xbp7ckzv42vkm008",
        name: "build",
        createdAt: "2024-05-28T15:32:19.489Z",
        status: "SUCCESS",
        duration: 9206,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #2",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "e06fkimhve7zj2917movmvsl",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-28T13:47:55.710Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 53745,
    jobs: [
      {
        id: "r05yucwc1s2kfkyt3vlo4oy7",
        name: "test",
        createdAt: "2024-05-28T13:47:55.710Z",
        status: "SUCCESS",
        duration: 19386,
        __typename: "Job",
      },
      {
        id: "e9y0s2p15nzat75jt0518rc0",
        name: "build",
        createdAt: "2024-05-28T13:47:55.710Z",
        status: "SUCCESS",
        duration: 33873,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #1",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
];
