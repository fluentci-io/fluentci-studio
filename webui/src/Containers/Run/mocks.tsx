export const actions: {
  id: string;
  name: string;
  duration: string;
  status: "SUCCESS" | "FAILURE" | "RUNNING" | "PENDING";
}[] = [
  {
    id: "1",
    name: "Test",
    duration: "5s",
    status: "SUCCESS",
  },
  {
    id: "1",
    name: "Build",
    duration: "3m 46s",
    status: "RUNNING",
  },
  {
    id: "1",
    name: "Deploy",
    duration: "7m 53s",
    status: "PENDING",
  },
];
