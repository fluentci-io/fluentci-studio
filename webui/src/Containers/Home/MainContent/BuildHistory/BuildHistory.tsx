import { FC, useMemo } from "react";
import { CheckCircle, Clock } from "@styled-icons/bootstrap";
import { CloseOutline } from "@styled-icons/evaicons-outline";
import { Bar, Bars, Info, Status, Unit, Value, Container } from "./styles";

export type BuildHistoryProps = {
  builds: {
    duration: number;
    status: string;
  }[];
  status: string;
  reliability: number;
  speed: number;
  buildsPerWeek: number;
};

const BuildHistory: FC<BuildHistoryProps> = (props) => {
  const { builds, status, reliability, speed, buildsPerWeek } = props;

  const max = useMemo(() => {
    return Math.max(...builds.map(({ duration }) => duration));
  }, [builds]);

  const _builds = useMemo(
    () =>
      builds.map(({ duration, status }) => ({
        duration: (duration * 100) / max,
        status,
      })),
    [builds, max]
  );

  const convertSpeed = (speed: number) => {
    if (speed < 60) return speed;
    if (speed < 3600) return Math.floor(speed / 60);
    return Math.floor(speed / 3600);
  };

  const convertSpeedUnit = (speed: number) => {
    if (speed < 60) return "s";
    if (speed < 3600) return "m";
    return "h";
  };

  return (
    <Container>
      <Bars>
        <div style={{ position: "relative", width: 179 }}>
          {_builds.map(({ duration, status }, i) => (
            <Bar max={duration} n={i + 1} success={status === "SUCCESS"} />
          ))}
        </div>
      </Bars>
      <Info>
        <div>Speed</div>
        <Value>
          {convertSpeed(Math.round((speed / 1000) * 100) / 100)}
          <Unit>{convertSpeedUnit(speed / 1000)}</Unit>
        </Value>
      </Info>
      <Info>
        <div>Reliability</div>
        <Value>
          {Math.round(reliability * 100) / 100}
          <Unit>%</Unit>
        </Value>
      </Info>
      <Info>
        <div>Builds</div>
        <Value>
          {buildsPerWeek}
          <Unit>/ week</Unit>
        </Value>
      </Info>
      <Status status={status}>
        {status === "SUCCESS" && <CheckCircle size="24" color="#05f076" />}
        {status === "PENDING" && <Clock size={"24"} color="orange" />}
        {status === "FAILURE" && <CloseOutline size={"24"} color="#ff006a" />}
      </Status>
    </Container>
  );
};

export default BuildHistory;
