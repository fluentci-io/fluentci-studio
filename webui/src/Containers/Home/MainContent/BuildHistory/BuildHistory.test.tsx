import { render } from "@testing-library/react";
import BuildHistory from "./BuildHistory";
import _ from "lodash";
import { projects } from "../mocks";

describe("BuildHistory", () => {
  it("should render the component", () => {
    const item = projects[0];
    const { container } = render(
      <BuildHistory
        status={_.last(item.recentRuns)?.status || "PENDING"}
        reliability={item.reliability || 0}
        speed={item.speed || 0}
        buildsPerWeek={item.buildsPerWeek || 0}
        builds={
          Array.from(Array(18).keys()).map((i) => ({
            status: _.get(item, `recentRuns.${i}.status`, ""),
            duration: _.get(item, `recentRuns.${i}.duration`, 0),
          })) || []
        }
      />
    );
    expect(container).toMatchSnapshot();
  });
});
