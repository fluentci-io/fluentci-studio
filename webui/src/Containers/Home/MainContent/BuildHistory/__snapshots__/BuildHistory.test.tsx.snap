// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BuildHistory > should render the component 1`] = `
<div>
  <div
    class="css-mxqn1d"
  >
    <div
      class="css-12j1nmi"
    >
      <div
        style="position: relative; width: 179px;"
      >
        <div
          class="css-15vwjnl"
          max="13.698697833016233"
        />
        <div
          class="css-19zlhna"
          max="6.232446278405114"
        />
        <div
          class="css-1hm00j7"
          max="7.233986475912793"
        />
        <div
          class="css-1je50oj"
          max="100"
        />
        <div
          class="css-1rckjuv"
          max="38.15901097905496"
        />
        <div
          class="css-5uuauc"
          max="15.191948143938458"
        />
        <div
          class="css-1g90bfm"
          max="5.592481859438446"
        />
        <div
          class="css-148582a"
          max="0"
        />
        <div
          class="css-yyu4jy"
          max="0"
        />
        <div
          class="css-1fqssq5"
          max="0"
        />
        <div
          class="css-klpgw9"
          max="0"
        />
        <div
          class="css-1crqmc2"
          max="0"
        />
        <div
          class="css-1xyk97i"
          max="0"
        />
        <div
          class="css-1pdbres"
          max="0"
        />
        <div
          class="css-1ciawv4"
          max="0"
        />
        <div
          class="css-14s0ibw"
          max="0"
        />
        <div
          class="css-1vmcux9"
          max="0"
        />
        <div
          class="css-1hj9s5a"
          max="0"
        />
      </div>
    </div>
    <div
      class="css-1j8ni2y"
    >
      <div>
        Speed
      </div>
      <div
        class="css-1o76smw"
      >
        32.28
        <span
          class="css-n3629o"
        >
          s
        </span>
      </div>
    </div>
    <div
      class="css-1j8ni2y"
    >
      <div>
        Reliability
      </div>
      <div
        class="css-1o76smw"
      >
        71.43
        <span
          class="css-n3629o"
        >
          %
        </span>
      </div>
    </div>
    <div
      class="css-1j8ni2y"
    >
      <div>
        Builds
      </div>
      <div
        class="css-1o76smw"
      >
        2
        <span
          class="css-n3629o"
        >
          / week
        </span>
      </div>
    </div>
    <div
      class="css-1gtb97y"
    >
      <svg
        aria-hidden="true"
        class="StyledIconBase-sc-ea9ulj-0 jvZCGA"
        color="#05f076"
        fill="currentColor"
        focusable="false"
        height="24"
        viewBox="0 0 16 16"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
        />
        <path
          d="M10.97 4.97a.235.235 0 0 0-.02.022L7.477 9.417 5.384 7.323a.75.75 0 0 0-1.06 1.06L6.97 11.03a.75.75 0 0 0 1.079-.02l3.992-4.99a.75.75 0 0 0-1.071-1.05z"
        />
      </svg>
    </div>
  </div>
</div>
`;
